src/abstracts/SablierV2Lockup.sol

```diff
@@ -277,14 +281,29 @@ abstract contract SablierV2Lockup is
             revert Errors.SablierV2Lockup_Overdraw(streamId, amount, withdrawableAmount);
         }
 
+        // Retrieve the sender from storage.
+        address sender = getSender(streamId);
+
         // Effects and Interactions: make the withdrawal.
         _withdraw(streamId, to, amount);
 
         // Interactions: if `msg.sender` is not the recipient and the recipient is a contract, try to invoke the
         // withdraw hook on it without reverting if the hook is not implemented, and also without bubbling up
         // any potential revert.
         if (msg.sender != recipient && recipient.code.length > 0) {
-            try ISablierV2LockupRecipient(recipient).onStreamWithdrawn({
+            try ISablierV2Recipient(recipient).onLockupStreamWithdrawn({
+                streamId: streamId,
+                caller: msg.sender,
+                to: to,
+                amount: amount
+            }) { } catch { }
+        }
+
+        // Interactions: if `msg.sender` is not the sender and the sender is a contract, try to invoke the
+        // withdraw hook on it without reverting if the hook is not implemented, and also without bubbling up
+        // any potential revert.
+        if (!isCallerStreamSender && sender.code.length > 0) {
```

Comment:
Since `isCallerStreamSender` will no longer be needed after this [PR: make withdraw callable by any account](https://github.com/sablier-labs/v2-core/pull/785/files#diff-aa886876392e91852d49ed143fbd48c2fd40b518b994039251f0da369c3cbeb1L249), I suggest we should use `msg.sender != sender` here.

Reply:
correct, i agree